### PR TITLE
optionally drop case actions and xform IDs from ES index

### PR DESCRIPTION
**Pull request by snopoke**
As far as I can tell these ES fields are not used in any critical way other than in the Case API.

### CODE REVIEW

**Comment by emord** [comment]
@snopoke I updated the doc in a comment, but actions are used by the Worker Activity Report https://github.com/dimagi/commcare-hq/blob/085e8aed4e253739aa381fde674c0106f364fd54/corehq/apps/es/cases.py#L90

File: corehq/pillows/case.py [hunk 1]
```diff
@@ -41,6 +41,12 @@ def transform_case_for_elasticsearch(doc_dict):
     if 'backend_id' not in doc_ret:
         doc_ret['backend_id'] = 'couch'
 
+    if settings.CASE_ES_DROP_FORM_FIELDS:
+        # these fields are lazily loaded to replacing or removing them
```
Comment (proteusvacuum) [inline corehq/pillows/case.py]: nit: `to` -> `so`

**Review by proteusvacuum — APPROVED** [review]
Great! If we can get rid of all usages of `actions`, I think the `CaseSearch` index is then basically a drop-in replacement.

**Comment by snopoke** [comment]
> actions are used by the Worker Activity Report

:-( I don't see a way around that without fundamentally changing how that report works.

I did a few tests and restricting the query results to the last 6 months or so improves the performance quite a bit (for cases with lots of transactions). I also checked the nginx logs and found that we have very few queries past the beginning of the year:

Date | query count
-- | --
2000-1 | 4
2010-1 | 2
2015-3 | 4
2016-1 | 1
2017-1 | 3
2018-10 | 3
2018-11 | 2
2018-12 | 3
2018-5 | 4
2018-7 | 6
2018-8 | 2
2019-1 | 341
2019-2 | 500
2019-3 | 80
2019-4 | 452
2019-5 | 6474
2019-6 | 7732
2019-7 | 2584

We'd need to check with other teams before we could make this change though. Closing for now.

**Comment by snopoke** [comment]
Got the go-ahead to make this change to the Worker Activity Report.

**Review by emord — APPROVED** [review]

**Comment by snopoke** [comment]
Tested this on staging with and without the setting.

**Comment by emord** [comment]
damn

image

**Pull request merged**